docs/infrastructure/microk8s/install.md

```diff
@@ -0,0 +1,274 @@
+4. After runing these commands, we have ensured that dns, http, and nginx ingress are enabled and working properly inside the cluster.
+
+### Installing Codacy
+
+#### Helm and Tiller
+
+1. First we must install `helm` onto the cluster. With `helm`, we can easily install charts and manage the lifecycle of the installed artifacts. One example is the ability to rollback a failed install.
+2. Assuming you would want to install `helm` version 2.16.3, export the following environment variable: `export HELM_VERSION=2.16.3`
+3. Run this script that will retrieve the desired `helm` version and install it onto the machine's `/usr/local/bin`: `HELM_PKG=helm-${HELM_VERSION}-linux-amd64.tar.gz
+wget https://get.helm.sh/$HELM_PKG
+tar xvzf $HELM_PKG
+mv linux-amd64/tiller linux-amd64/helm /usr/local/bin
+rm -rvf $HELM_PKG linux-amd64/`.
+   1. Two executables will get installed: `helm` and `tiller`. The former is responsible for resolving the configuration of the chart to be installed, while issuing the correct install commands onto the cluster. The latter is responsible for receiving the install commands issued by `helm`, as well as managing the lifecycle of the components that have been installed. `helm` is the client facing side, while `tiller` is the server/cluster facing side.
+4. We must now deploy `tiller` onto the cluster.
+   1. First we must set a `serviceaccount` and a `clusterrolebinding` for `tiller` on the microk8s cluster: `microk8s.kubectl create serviceaccount --namespace kube-system tiller
+microk8s.kubectl create clusterrolebinding tiller-cluster-rule --clusterrole=cluster-admin --serviceaccount=kube-system:tiller`
+   2. Then we can install `tiller`, which will be installed with the same `HELM_VERSION` that you have specified earlier: `helm init --service-account tiller`
+   3. We wait for tiller to be ready: `microk8s.kubectl -n kube-system wait --for=condition=Ready pod -l name=tiller --timeout=300s`
+
+### Codacy chart
+
+1. Create a namespace inside the cluster onto which codacy will be deployed. `microk8s.kubectl create namespace codacy`
+2. The microk8s cluster must know some docker credentials so that the container docker images can be pulled onto the pods. These credentials must belong to the namespace where codacy is installed. You can do so by running: `microk8s.kubectl create secret docker-registry docker-credentials --docker-username=${DOCKER_USERNAME} --docker-password=${DOCKER_PASSWORD} --namespace codacy`. These credentials will be encoded inside the cluster.
+3. For codacy to work, we must set a file that which contains information like how many replicas of a given pod will codacy have, what are the cpu and memory limits for these pods, which secret tokens to use for the filestore, among others.
+   1. Define the following variables
+      1. `CODACY_PROTOCOL` - http or https protocol for your codacy installation.
+      2. `CODACY_URL` - desired URL for your codacy installation.
+      3. `CRYPTO_SECRET` - A secret...
```

Comment:
We should probably give guidance on how to generate such secret (e.g. min length etc.)